File: client/src/app/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  imageUrl: string | null;
}

export interface Pagination<T> {
  page: number;
  pageSize: number;
  totalCount: number;
  data: T[];
}

export interface ProductFilters {
  brands: string[];
  categories: string[];
}

export interface ProductQueryParams {
  pageIndex: number;
  pageSize: number;
  search: string;
  brand: string;
  category: string;
  minPrice: number | null;
  maxPrice: number | null;
  sort: string;
}

File: api/Models/Product.cs
namespace Skinet.Api.Models;

public class Product
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public int StockQuantity { get; set; }

    public string? ImageUrl { get; set; }  

    public List<OrderItem> OrderItems { get; set; } = new();

    public string Brand { get; set; } = string.Empty;
    
    public string Category { get; set; } = string.Empty;

}
